apps/frontend/core-dapp/src/components/TorusTrap.tsx

```diff
@@ -0,0 +1,93 @@
+import { FC, Suspense, useEffect, useState } from 'react'
+import { useIntercom } from './Intercom'
+
+export function useDetectTorusIframe(): HTMLIFrameElement | undefined {
```

Comment:
While I prefer having the logics in both hooks combined into 1 hook rather than making 2 hooks, if you prefer this approach, we should not export `useDetectTorusIframe` as it is not useful anywhere else
```suggestion
function useDetectTorusIframe(): HTMLIFrameElement | undefined {
```

Reply:
Good catch, removed `export` from both. You're right I'd rather keep them separate as I think their purpose is different (one detects the iframe presence, other detects the iframe size)